src/core/config/transformer/union_input_type.rs

```diff
@@ -0,0 +1,108 @@
+use std::borrow::Cow;
+
+use super::Transform;
+use crate::core::config::{Arg, Config, Field, Type};
+use crate::core::valid::Valid;
+
+#[derive(Default)]
+pub struct UnionInputType;
+
+impl Transform for UnionInputType {
```

Comment:
Can we split this PR into two — One with just the transformer and one with the the from_proto changes.

Reply:
updated this pr and created https://github.com/tailcallhq/tailcall/pull/2221